private-templates-service/client/src/components/Common/SpinnerModal/SpinnerModal.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { CenteredSpinner } from './styled';
+import { connect } from 'react-redux';
+import { RootState } from '../../../store/rootReducer';
+import { closeModal } from '../../../store/page/actions';
+import { BackDrop } from '../ShareModal/styled';
+import { SpinnerSize } from 'office-ui-fabric-react';
+
+const mapStateToProps = (state: RootState) => {
+  return {
+    isFetching: state.currentTemplate.isFetching
+  }
+}
+
+const mapDispatchToProps = (dispatch: any) => {
+  return {
+    closeModal: () => {
+      dispatch(closeModal());
+    }
+  }
+}
+
+interface Props {
+  isFetching: boolean;
+  closeModal: () => void;
+  closeAction?: () => void;
+}
+
+class SpinnerModal extends React.Component<Props>{
+  constructor(props: Props) {
+    super(props);
+  }
+  
+  componentDidUpdate(){
+    if(!this.props.isFetching){
+      if(this.props.closeAction){
+        this.props.closeAction();
+      }
+      this.props.closeModal();
```

Comment:
I agree with what Kody said in his comments, although we treat this component like a Modal it really isn't, it's just an overlay for when `isFetching === true`. I feel like the `openModal()` and `closeModal()` actions aren't needed in this case.

Reply:
Also that implies we don't need the `ModalState` enum